asm/f64/benchAxpy_test.go

```diff
@@ -0,0 +1,167 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build go1.7
+
+package f64
+
+import (
+	"fmt"
+	"testing"
+)
+
+const (
+	testLen = 1000000
+)
+
+var (
+	a = float64(2)
+	x = make([]float64, testLen)
+	y = make([]float64, testLen)
+	z = make([]float64, testLen)
+)
+
+func init() {
+	for n := range x {
+		x[n] = float64(n)
+		y[n] = float64(n)
+	}
+}
+
+func BenchmarkAxpyUnitary(t *testing.B) {
+	naiveaxpyu := func(a float64, x, y []float64) {
+		for i, v := range x {
+			y[i] += a * v
+		}
+	}
+	tests := []struct {
+		name string
+		f    func(a float64, x, y []float64)
+	}{
+		{"AxpyUnitary", AxpyUnitary},
+		{"NaiveAxpyUnitary", naiveaxpyu},
+	}
+	for _, test := range tests {
+		for _, ln := range []uintptr{1, 3, 10, 30, 1e2, 3e2, 1e3, 3e3, 1e4, 3e4} {
+			t.Run(fmt.Sprintf("%s-%d", test.name, ln), func(b *testing.B) {
+				b.SetBytes(int64(64 * ln))
+				x, y := x[:ln], y[:ln]
+				b.ResetTimer()
+				for i := 0; i < b.N; i++ {
+					test.f(a, x, y)
```

Comment:
`y` could overflow, but that shouldn't change the operation timing.  The optimization manual only notes value-dependent latency changes for `DIVPD/DIVSD` and `SQRTPD/SQRTSD` instructions.